### PR TITLE
ocm observability add publish

### PR DESCRIPTION
**Pull request by mqhackett**
adds publish jobs and creates release 2.4 config

### CODE REVIEW

File: ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.3.yaml [hunk 1]
```diff
@@ -21,7 +21,6 @@ images:
       - registry.ci.openshift.org/open-cluster-management/builder:go1.16-linux
   to: insights-client
 promotion:
-  disabled: true
```
Comment (jlpadilla) [inline ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.3.yaml:24]: Why this change?  We only promote from the main branch.

File: ci-operator/config/open-cluster-management/must-gather/open-cluster-management-must-gather-release-2.4.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+base_images:
+  cli:
+    name: "4.7"
```
Comment (jlpadilla) [inline ci-operator/config/open-cluster-management/must-gather/open-cluster-management-must-gather-release-2.4.yaml:3]: I expect that `4.8` is now available.  Let's also update main.

File: ci-operator/config/open-cluster-management/must-gather/open-cluster-management-must-gather-release-2.4.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    name: builder
+    namespace: open-cluster-management
+    tag: go1.16-linux-amd64
+images:
+- dockerfile_path: Dockerfile.prow
+  from: cli
+  inputs:
+    cli:
+      as:
+      - registry.ci.openshift.org/ocp/4.7:cli
```
Comment (jlpadilla) [inline ci-operator/config/open-cluster-management/must-gather/open-cluster-management-must-gather-release-2.4.yaml:17]: Same here, let's move to `4.8` to avoid problems later.

File: ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.3.yaml (hunk 1 above)
Comment (mqhackett) [reply ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.3.yaml:24]: The disabled label exists in the current release branch. The instructions said to remove that line from the previous release. https://github.com/open-cluster-management/cicd-docs/blob/main/prow/new_release.md#update-the-prow-config-file-for-the-previous-release-branch

Comment (jlpadilla) [reply ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.3.yaml:24]: ok, let's follow the instructions, maybe I don't understand this field well enough.

File: ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.4.yaml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  disabled: true
+  name: "2.4"
+  namespace: open-cluster-management
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: publish
```
Comment (jlpadilla) [inline ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.4.yaml:33]: @mqhackett We need to disable publish from travis before we can merge this.

File: ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.4.yaml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    workflow: ocm-ci-image-mirror
+- as: backplane-fast-forward
+  postsubmit: true
+  steps:
+    env:
+      DESTINATION_BRANCH: release-2.4
+    workflow: ocm-ci-fastforward
```
Comment (jlpadilla) [inline ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.4.yaml:63]: Only the `main` branch should do the fast-foward.
```suggestion
```

File: ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.4.yaml (hunk 4 above)
Comment (mqhackett) [reply ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.4.yaml:33]: @jlpadilla Here is a PR to disbable for 

must-gather: https://github.com/open-cluster-management/must-gather/pull/70

insights-client: https://github.com/open-cluster-management/insights-client/pull/55

Comment (jlpadilla) [reply ci-operator/config/open-cluster-management/insights-client/open-cluster-management-insights-client-release-2.4.yaml:33]: Thanks, both PRs are merged.  We can merge this one now.

**Comment by jlpadilla** [comment]
@mqhackett Some checks are failing.  You need to run the following before committing changes.
```
make prow-config && make update
```
